test/one_to_one.jl

```diff
@@ -71,10 +65,13 @@ function read_data(a::CaloHitContributionCollection)
 	@assert sccon2.stepPosition == Vector3d{Float32}(7, 6.32, 12)
 	@assert sccon2.particle.PDG == 2212
 	@assert sccon2.particle.mass == 0.938
-	@assert sccon2.particle.momentumAtEndpoint == Vector3d{Float32}(0.0,0.0,-7000.0)
+	@assert sccon2.particle.momentumAtEndpoint == Vector3d{Float32}(0.0,0.0,7000.0)
 	@assert sccon2.particle.generatorStatus == 3
 	@assert sccon2.energy == Float32(13.6)
 
+	# checking egality of particle object
+	@assert sccon2.particle === sccon1.particle
```

Comment:
This essentially already encompasses what we want to know. It is not yet entirely clear to me if we can manage this in the templates as well. Could you add a few explicit checks that compare a few properties of the particle with just the equality operator, i.e. something like
```julia
@assert sccon2.particle.PDG == sccon1.particle.PDG
```